File: tp1_2_3/java/main/java/heritage/Enseignant.java
package heritage;

public class Enseignant extends Personne {
    private double salaire;

    public Enseignant(String nom, int age, double salaire) {
        super(nom, age);
        this.salaire = salaire;
    }

    public double getSalaire() {
        return salaire;
    }

    @Override
    public void afficherDetails() {
        super.afficherDetails();
        System.out.println("Salaire : " + salaire + " EUR");
    }
}

File: tp1/python/src/enseignant.py
from src.personne import Personne


class Enseignant(Personne):
    def __init__(self, nom, age, salaire):
        super().__init__(nom, age)
        self.__salaire = salaire

    @property
    def salaire(self):
        return self.__salaire

    def afficher_details(self):
        super().afficher_details()
        print(f"Salaire : {self.__salaire} EUR")
